### ISSUE TITLE
Registry 500-out

### ISSUE THREAD

**Issue by dmp42** [docker/distribution#100]
Using samalba/docker-registry and pushing using 1.5RC I can trigger 500 consistently.

Unfortunately, there is very little information available in the registry logs save the access log itself:

```
192.168.100.68 - - [23/Jan/2015:18:03:42 +0000] "PUT /v2/dmp42/testthing/blobs/uploads/86925eee-5116-4101-9921-73df5e9b7de7?_state=OhwhFPtmp1PHSaoeklO8z2yiwS5-OiMYphRR7TG8DNh7Ik5hbWUiOiJkbXA0Mi90ZXN0dGhpbmciLCJVVUlEIjoiODY5MjVlZWUtNTExNi00MTAxLTk5MjEtNzNkZjVlOWI3ZGU3IiwiT2Zmc2V0IjowLCJTdGFydGVkQXQiOiIyMDE1LTAxLTIzVDE4OjAzOjQxLjU3OTkzMjgzOFoifQ%3D%3D&digest=tarsum.v1%2Bsha256%3A974f176a8980209133a39750d7a3fd06f3addd572303e41819a38b089b89b154 HTTP/1.1" 500 84 "" "docker/1.5.0 go/go1.4.1 git-commit/7e803ba kernel/3.8.0-29-generic os/linux arch/amd64" 
```

**Comment by stevvooe** [comment]
@dmp42 Could you describe the conditions that reproduce this? It's possible this is an issue with s3, so please try on a local registry and we'll go from there.

**Comment by stevvooe** [comment]
I think this may be addressed by #121. If you see this again, please report.
